Sources/DatadogSDKTesting/FrameworkLoadHandler.swift

```diff
@@ -31,18 +34,25 @@ public class FrameworkLoadHandler: NSObject {
             environment["XCTestBundlePath"] != nil ||
             environment["SDKROOT"] != nil
         if isInTestMode {
-            guard DDEnvironmentValues.getEnvVariable("DATADOG_CLIENT_TOKEN") != nil || DDEnvironmentValues.getEnvVariable("DD_API_KEY") != nil else {
-                print("[DatadogSDKTesting] DATADOG_CLIENT_TOKEN or DD_API_KEY are missing.")
-                return
+            if !DDTestMonitor.tracer.isBinaryUnderUITesting {
+                if !DDTestMonitor.env.disableTestInstrumenting {
+                    DDTestObserver().startObserving()
```

Comment:
Why aren't we retaining the test observer instance? 🤔 In downstream logic it will be retained anyway in:
```swift
XCTestObservationCenter.shared.addTestObserver(self)
```
so why does `FrameworkLoadHandler` not own it in this condition?

Reply:
I was an oversight on my side when changing this code, but as you say in the next comment this code is very confusing, I will refactor as you say